9 Kms to Miles

leggant opened the pull request:
## Items completed
1. created input form, with radio button fields
2. added to the site main header component
3. logic handled within a DistanceContext hook
   - Context wraps top level app component 
   - context, useDistanceValues hook created individually
4. added prettier code formatting config files and dependency
5. Components with `.js` extensions have been switched to compatible `.jsx` extensions. 
6. Import statements have file extensions from from file paths

## Bug fixes
1. node packages had security warnings
    - found these to be caused by `react-scripts` package
    - discovered that the app was created using the deprecated `create-react-app` command
 2. reconfigured the app with _vite_ ([the current recommended method](https://react.dev/blog/2025/02/14/sunsetting-create-react-app))
    - updated app package.json, removing deprecated packages, installing _vite_ related dependencies and scripts
    - updated the `.envexample` file client keys so these can be accessible
    - **Note: `.env` files need to be updated to have the new key name**
    - `index.html` moved to root directory - required by _vite_
    - `vite.config.js`, `eslint.config.js` added to project root.

## To be fixed later
1. remove the useLocalStorage hook so the users distanceType ('miles', 'kilometers') is not saved to local storage, but to their profile in the database.

closes #9 
closes #34

leggant committed:
add author and engine keys

set standard for node compatibility

leggant committed:
update dependencies. remove react-scripts due to vulnerablities

installed firebase hooks package to assist with firebase actions

leggant committed:
update package.json

add additional config keys

leggant committed:
create `src\utils\constants.js`

export site wide variables

leggant committed:
create `src\utils\data-conversion.js`

export distance conversion util functions. Added jsdoc annotations

leggant committed:
add imported constants to `data-conversion.js` functions

leggant committed:
#9 update constant names

for distance conversion

leggant committed:
#9 update constant file

add commented sections

leggant committed:
#9 add `prettier` config

install dev dependency `package.json`

leggant committed:
`src\utils\constants.js` add constants

for distance reducer function

#9

leggant committed:
#9 create distance checkbox form component

has a prop named `kilometersOn` this will need to be passed down from a context hook/global state provider

leggant committed:
#9 add memo hook to checkbox form component

leggant committed:
#9 update distance constants

removed reducer object as this is not needed

leggant committed:
#9 create distance context hook

leggant committed:
#9 refactor distance context to include default values

leggant committed:
#9 add distance context provider to top level app

leggant committed:
#9 create local storage hook

to be used to add key value for distance setting in the browser

leggant committed:
#9 update distance checkbox form component

add additional props for state and functions passed from the outer component

leggant committed:
#9 prettify distance checkbox form component

leggant committed:
#9 prettify distance checkbox form component

leggant committed:
reconfigure app to use vite inplace of deprecated react configuration

components renamed with `.jsx` extensions

leggant committed:
#9 fixed error loading env values for the client

vite requires the `VITE_...` keys to load into the client

leggant committed:
#9 reconfigured/renamed distance form

replaced checkbox input with radio buttons

leggant committed:
#9 add distance radio button component to the header

leggant committed:
#9 seperate distance context into individual files

update import into `App.jsx`

leggant committed:
#9 add state import into `useLocalStorage` hook

leggant committed:
#9 convert hook file to `.tsx` extension

leggant committed:
#9 update useLocalStorage hook

still not working in testing

leggant committed:
#9 fix use local storage hook

add to distance radio button form

leggant committed:
#9 install zod input validation package

to handle type checking custom hook inputs

HaydenMCD commented on package.json:
```diff
@@ -1,20 +1,42 @@
+  },
+  "author": {
+    "name": "Hayden McDowall",
+    "email": "haysay424@gmail.com"
```
Remove email and name.

HaydenMCD commented on src/utils/data-conversion.js:
Create a util for calculating day etc

leggant replied:
created new file in utils directory. removed the static math file replacing it with date-conversion.js

leggant commented on src/components/form/DistanceRadioButtonForm.jsx:
```diff
@@ -0,0 +1,50 @@
+import { useLocalStorage } from '../../hooks/useLocalStorage';
+/**
+ *
+ * @file DistanceCheckBoxForm.js
```
Change made.

leggant committed:
#9 remove name object from package.json

leggant committed:
#9 create date conversion utils

update constants.js and remove staticmath.js. replace utils from this with utils in the date conversion file

leggant committed:
#9 correct filename jsdoc declaration

leggant committed:
#9 correct component import statement

leggant committed:
#9 update component to use constant.js

format file with prettier

leggant committed:
#9 update component to use date utils

replacing file path. format file using prettier

leggant committed:
#9 move context to context directory

update related import statements. Rename `useDistanceValues` hook to have .js extenstion
